groups/reconcile.go

```diff
@@ -0,0 +1,399 @@
+				} else {
+					fmt.Printf("Trying to create group: %s\n", groupEmailId)
+					g4, err := srv.Groups.Insert(&admin.Group{
+						Email:       groupEmailId,
+						Name:        description,
+						Description: "Kubernetes wg-k8s-infra test group #2",
+					}).Do()
+					if err != nil {
+						log.Fatalf("Unable to add new group %s: %v", groupEmailId, err)
+					} else {
+						fmt.Printf("> Successfully created group %s\n", g4.Email)
+					}
+				}
+			} else {
+				log.Fatalf("Unable to fetch group: %#v", err)
+			}
+		default:
+			log.Fatalf("Unable to fetch group: %#v", err.Error())
+		}
+	}
+}
+
+func deleteGroupsIfNecessary(service *admin.Service) {
+	g, err := service.Groups.List().Customer("my_customer").OrderBy("email").Do()
+	if err != nil {
+		log.Fatalf("Unable to retrieve users in domain: %v", err)
+	}
+	if len(g.Groups) == 0 {
+		fmt.Print("No groups found.\n")
+	} else {
+		for _, g := range g.Groups {
+			// Don't touch existing mailing lists, we should
+			// always prefix with "k8s-infra-"
+			if !strings.HasPrefix(g.Email, "k8s-infra-") {
+				continue
+			}
+			found := false
+			for _, g2 := range groupsConfig.Groups {
+				if g2.EmailId == g.Email {
+					found = true
+					break
+				}
+			}
+			if found {
+				continue
+			}
+			// We did not find the group in our groups.xml, so delete the group
+			if config.DryRun {
+				fmt.Printf("dry-run : Skipping removing group %s\n", g.Email)
+			} else {
+				err := service.Groups.Delete(g.Email).Do()
+				if err != nil {
+					log.Fatalf("Unable to remove group %s : %v", g.Email, err)
+				} else {
+					fmt.Printf("Removing group %s\n", g.Email)
+				}
+			}
+
+		}
+	}
+}
+
+func updateGroupSettingsToAllowExternalMembers(srv *groupssettings.Service, groupEmailId string) {
+	g2, err := srv.Groups.Get(groupEmailId).Do()
+	if err != nil {
+		switch err := err.(type) {
+		case *googleapi.Error:
+			if err.Code == http.StatusNotFound {
+				fmt.Printf("skipping updating group settings as group %s has not yet been created\n", groupEmailId)
+				return
+			} else {
+				log.Fatalf("Unable to retrieve group info for group %s: %v", groupEmailId, err)
+			}
+		default:
+			log.Fatalf("Unable to retrieve group info for group %s: %v", groupEmailId, err)
+		}
+	}
+	if g2.AllowExternalMembers != "true" {
+		if config.DryRun {
+			fmt.Printf("dry-run : skipping updating group settings\n")
+		} else {
+			_, err := srv.Groups.Patch(groupEmailId, &groupssettings.Groups{
+				AllowExternalMembers: "true",
+			}).Do()
+			if err != nil {
+				log.Fatalf("Unable to update group info for group %s: %v", groupEmailId, err)
+			} else {
+				fmt.Printf("> Successfully updated group settings for %s to allow external members\n", groupEmailId)
+			}
+		}
+	}
+}
+
+func addMembersToGroup(service *admin.Service, groupEmailId string, members []string) {
+	l, err := service.Members.List(groupEmailId).Do()
+	if err != nil {
+		switch err := err.(type) {
```

Comment:
Maybe change this switch to:
```
if apierr, ok := err.(*googleapi.Error); ok && err.Code == http.StatusNotFound {
  fmt.Printf("skipping adding members to group %s as it has not yet been created\n", groupEmailId)
  return
}
log.Fatalf("Unable to retrieve members in group %s: %v", groupEmailId, err)
```

Reply:
Fixed. thanks for the tip to make it better!